Terminal.Gui/Input/Responder.cs

```diff
@@ -147,31 +115,61 @@ public virtual bool ProcessKey (KeyEvent keyEvent)
 		///  </para>
 		/// </remarks>
 		/// <param name="keyEvent">Contains the details about the key that produced the event.</param>
-		public virtual bool ProcessColdKey (KeyEvent keyEvent)
+		public virtual bool OnColdKey (KeyEventArgs keyEvent)
 		{
 			return false;
 		}
 
 		/// <summary>
-		/// Method invoked when a key is pressed.
+		/// If the view is focused, gives the view a chance to process the keystroke.
+		/// Called after <see cref="OnKeyDown"/> and <see cref="OnKeyUp"/>.
+		/// Fires the <see cref="KeyPressed"/> event.
+		/// Typical apps will use <see cref="Command"/> instead.
 		/// </summary>
 		/// <param name="keyEvent">Contains the details about the key that produced the event.</param>
+		/// <returns><see langword="false"/> if the key stroke was not handled. <see langword="true"/> if no
+		/// other view should see it.</returns>
+		public virtual bool OnKeyPressed (KeyEventArgs keyEvent)
+		{
+			// fire event
+			KeyPressed?.Invoke (this, keyEvent);
+			return keyEvent.Handled;
```

Comment:
This will always the last to been running. So if a user subscribe `KeyPressed` event and this method isn't hit, the user never will handle the event. I understand your concern, maybe you tough that this is the first to be executed. But isn't and maybe it's never called if an overridden method returns true or `KeyPressed` event return `Handled` is true. This is only called if `base` is called.

Reply:
It's better to reset it to how it was previously:

```cs
public virtual bool OnKeyPressed (KeyEventArgs keyEvent)
{
	return false;
}
```

Reply:
For your code to work properly users must subscribing directly from a `Responder` class when normally the user will use `View` class. If a user want to cancel a key press for a view it's enough subscribing `view.KeyPressed`. I think that not make sense. So I propose to change this unit test:

```cs
		[Fact]
		public void KeyPressed_Handled_True_Cancels_KeyPress ()
		{
			var v = new View ();
			var args = new KeyEventArgs () { Key = Key.Unknown };

			Assert.False (v.OnKeyPressed (args));
			Assert.False (args.Handled);

			v.KeyPressed += (s, a) => a.Handled = true;
			Assert.True (v.OnKeyPressed (args));
			Assert.True (args.Handled);

			v.Dispose ();
		}

```